terraform/environments/bootstrap/single-sign-on/policies.tf

```diff
@@ -1330,3 +1330,26 @@ data "aws_iam_policy_document" "fleet-manager-document" {
     resources = ["*"]
   }
 }
+
+resource "aws_iam_policy" "s3_upload_policy" {
+  provider = aws.workspace
+  name     = "s3_upload_policy"
+  path     = "/"
+  policy   = data.aws_iam_policy_document.s3_upload_policy_document.json
+}
+
+#tfsec:ignore:aws-iam-no-policy-wildcards
+data "aws_iam_policy_document" "s3_upload_policy_document" {
+  statement {
+    sid    = "AllowS3Upload"
+    effect = "Allow"
+    actions = [
+      "s3:PutObject",
+      "s3:ListBucket"
+    ]
+    resources = [
+      "arn:aws:s3:::mod-platform-image-artefact-bucket20230203091453221500000001",
```

Comment:
I don't think this should be hard coded; I know it can be a pain to get information on bucket ARNs without resorting to hard coding, but hard coding ARNs is a very brittle approach. Do we even need to specify access to a single bucket?

Reply:
I agree, not a good practice. My feeling is that the user wants to narrow it down to that specific bucket though (Sukesh is to confirm that), but, rather than referencing the bucket itself in the IAM role policy, perhaps we can narrow it down at the s3 bucket policy (by listing principals to specific roles?)

Reply:
Discussed with Ewa. Since there are many roles assuming access to the destination bucket, it is a bit challenging to edit the S3 permissions. Replaced hardcoded ARNs with dynamically fetched ARNs.